From 973404644c98a1664f44dc15e522c1fa2e4b368c Mon Sep 17 00:00:00 2001
Message-Id: <973404644c98a1664f44dc15e522c1fa2e4b368c.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Arun R Murthy <arun.r.murthy@intel.com>
Date: Mon, 10 Mar 2014 14:26:27 +0530
Subject: [PATCH 098/312] FOR_UPSTREAM [VPG]: drm/i915: use hrtimer in wait
 for vblank.

In wait for vblank use usleep_range, which will use hrtimers instead of
msleep. Using msleep(1~20) there are more chances of sleeping for 20ms.
Using usleep_range uses hrtimers and hence are precise, worst case will
trigger an interrupt at the higher/max timeout.

Issue: GMIN-1501
Change-Id: I97b8290ceff8d37aeaffbab98daf3a352a29b56d
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_drv.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_drv.h b/drivers/gpu/drm/i915/intel_drv.h
index ad6d244..8fcf152 100644
--- a/drivers/gpu/drm/i915/intel_drv.h
+++ b/drivers/gpu/drm/i915/intel_drv.h
@@ -53,7 +53,7 @@
 			break;						\
 		}							\
 		if (W && drm_can_sleep())  {				\
-			msleep(W);					\
+			usleep_range(W*1000, W*2*1000);	\
 		} else {						\
 			cpu_relax();					\
 		}							\
-- 
1.7.9.5

